File: src/geometryService/cpp/stractNet/LongFrag.cpp
///////////////////////////////////////////////////////////
//  LongFrag.cpp
//  Implementation of the Class LongFrag
//  Created on:      06-Aug-2008 7:51:41 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////
#include "LongFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////
LongFrag::~LongFrag() {}
LongFrag::LongFrag(Long lo) {
	MsgFrag("LongFrag", lo, MsgFrag::mftLong);
}
LongFrag::LongFrag(DataInputStream& in) {
	MsgFrag("LongFrag", in, MsgFrag::mftLong);
}
///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////
void LongFrag::Deserialize(DataInputStream& in) {
	try {
		// Read Integer
		this->field = in.readLong();
	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return;
}

void LongFrag::Serialize_Specific(DataOutputStream& out) {
	try {
		// write in new data
		out.writeLong(this->field);
	} catch (IOException ioe) {
		SNRoot::err("IOException in Serialize()", 0);
	}
	return;
}

File: src/iBME/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/LongFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public class LongFrag extends MsgFrag<Long> {
	public LongFrag(Long lo) {
		super("LongFrag", lo, MsgFrag.mftLong);
	}

	public LongFrag(DataInputStream in) {
		super("LongFrag", in, MsgFrag.mftLong);
	}

	@Override
	public void Deserialize(DataInputStream in) {
		try {
			// Read Integer
			this.field = in.readLong();
		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			// write in new data
			out.writeLong(this.field);
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}

}
